apps/web/server/routers/viewer/slots.tsx

```diff
@@ -169,76 +163,39 @@ export const slotsRouter = createRouter().query("getSchedule", {
-    const endTime = dayjs(input.endTime);
+    const { startTime, endTime } = input;
     if (!startTime.isValid() || !endTime.isValid()) {
       throw new TRPCError({ message: "Invalid time range given.", code: "BAD_REQUEST" });
     }
 
+    let currentSeats: CurrentSeats | undefined = undefined;
+
     const userSchedules = await Promise.all(
       eventType.users.map(async (currentUser) => {
-        const schedule = currentUser.schedules.filter(
-          (schedule) => !currentUser.defaultScheduleId || schedule.id === currentUser.defaultScheduleId
-        )[0];
-
-        const workingHours = getWorkingHours(
+        const {
+          busy,
+          workingHours,
+          currentSeats: _currentSeats,
+        } = await getUserAvailability(
           {
-            timeZone: currentUser.timeZone,
+            userId: currentUser.id,
+            dateFrom: startTime.format(),
+            dateTo: endTime.format(),
+            eventTypeId: input.eventTypeId,
           },
-          schedule.availability || currentUser.availability
+          { user: currentUser, eventType, currentSeats }
         );
-
-        const busy = await getBusyTimes({
-          credentials: currentUser.credentials,
-          startTime: startTime.format(),
-          endTime: endTime.format(),
-          eventTypeId: input.eventTypeId,
-          userId: currentUser.id,
-          selectedCalendars: currentUser.selectedCalendars,
-        });
+        if (!currentSeats && _currentSeats) currentSeats = _currentSeats;
 
         return {
           workingHours,
-          busy: busy.map((a) => ({
-            start: dayjs(a.start).subtract(currentUser.bufferTime, "minute").toDate(),
-            end: dayjs(a.end).add(currentUser.bufferTime, "minute").toDate(),
-          })),
+          busy,
         };
       })
     );
-
-    const schedule = eventType?.schedule;
-    const timeZone = schedule?.timeZone || eventType?.timeZone || undefined;
-
-    const workingHours = getWorkingHours(
-      {
-        timeZone,
-      },
-      schedule?.availability || (eventType?.availability.length ? eventType.availability : [])
-    );
-
-    /* Current logic is if a booking is in a time slot mark it as busy, but seats can have more than one attendee so grab
-      current bookings with a seats event type and display them on the calendar, even if they are full */
-    const currentSeats = eventType?.seatsPerTimeSlot
-      ? await ctx.prisma.booking.findMany({
-          where: {
-            eventTypeId: input.eventTypeId,
-            startTime: {
-              gte: startTime.format(),
-              lte: endTime.format(),
-            },
-          },
-          select: {
-            uid: true,
-            startTime: true,
-            _count: {
-              select: {
-                attendees: true,
-              },
-            },
-          },
-        })
-      : [];
+    const workingHours = userSchedules.flatMap((s) => s.workingHours);
```

Comment:
This works, but caused an issue I'm working around in the simplest method I could think of: #3082